app/controllers/tracks_controller.rb

```diff
@@ -0,0 +1,21 @@
+class TracksController < ApplicationController
+  before_action :use_track, except: :index
+
+  def authenticated_index
+    @tracks = Track.all
+  end
+
+  def authenticated_show
+    @user_track = current_user.user_tracks.find_by(track: @track)
+  end
+
+  def join
```

Comment:
Don't you want to POST on `/tracks/ruby/membership` instead?

Reply:
Maybe. I dunno :shrug:

I don't mind controllers having responsibility for multiple interactions as long as they don't have a lot of logic in each action. I find having lots of controllers with single actions (such as a membership action here) can get really noisy very quickly so I tend to start with just the add-on action and then refactor it out into a second controller once I know I need more actions (eg the leave track or pause track here).

The other consideration for me here is that the track isn't editable in this context. The actual controller is all about a track and the user's relationship with it. Which means I'd be tempted to fill a few actions on here because it won't have the other CRUD things. 

I guess from a philosophical standpoint I care less about the standard REST actions in websites than I do in APIs.

Reply:
Oh yeah, as long as it's a conscious choice, that's fine 👍

I just figured you internally modelled it as "track membership".

Reply:
In my experience, it's easier for me to model this as separate controllers because sometimes the `before_action`s do too much.  This is okay for now, but I see it as separate controllers especially if we take into account errors when user joins a track.